Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+        )
+    }
+
+    public convenience init<C: Collection>(
+        dataChunks: C
+    ) where C.Element: StringProtocol {
+        self.init(
+            dataChunks: dataChunks.map(\.asBodyChunk)
+        )
+    }
+
+    public convenience init(
+        stream: AsyncThrowingStream<some StringProtocol, any Error>,
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
+
+    public convenience init(
+        stream: AsyncStream<some StringProtocol>,
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
+
+    public convenience init<S: AsyncSequence>(
+        sequence: S,
+        length: Body.Length,
+        iterationBehavior: IterationBehavior
+    ) where S.Element: StringProtocol {
+        self.init(
+            sequence: .init(sequence.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+}
+
+extension Body {
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes`, converts it to String, and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the body contains more
+    ///   than `maxBytes`.
+    public func collectAsString(upTo maxBytes: Int) async throws -> String {
+        let bytes: DataType = try await collect(upTo: maxBytes)
+        return String(decoding: bytes, as: UTF8.self)
+    }
+}
+
+// MARK: - Body conversions
+
+extension Body: ExpressibleByStringLiteral {
+
+    public convenience init(stringLiteral value: String) {
+        self.init(data: value)
+    }
+}
+
+extension Body {
+
+    public convenience init(data: [UInt8]) {
+        self.init(data: data[...])
+    }
+}
+
+extension Body: ExpressibleByArrayLiteral {
+
+    public typealias ArrayLiteralElement = UInt8
+
+    public convenience init(arrayLiteral elements: UInt8...) {
+        self.init(data: elements)
+    }
+}
+
+extension Body {
+
+    public convenience init(data: Data) {
+        self.init(data: ArraySlice(data))
+    }
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes`, converts it to Foundation.Data, and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the body contains more
+    ///   than `maxBytes`.
+    public func collectAsData(upTo maxBytes: Int) async throws -> Data {
+        let bytes: DataType = try await collect(upTo: maxBytes)
+        return Data(bytes)
+    }
```

Comment:
IMO this doesn't carry its weight for me since it is a one-liner for the user.

Reply:
It was meant to mirror `collectAsString` in a way, and tries to help with systems where Data is the very common data type (iOS apps). Plus, it'll be called from generated code, from where we really try to only make simple calls, as opposed to embedding too much logic.